pkg/agent/manager/manager_test.go

```diff
@@ -963,6 +963,98 @@ func TestSynchronizationWithLRUCache(t *testing.T) {
 	require.Equal(t, clk.Now(), m.GetLastSync())
 }
 
+func TestSyncRetriesWithDefaultIntervalOnZeroSVIDSReturned(t *testing.T) {
+	dir := spiretest.TempDir(t)
+	km := fakeagentkeymanager.New(t, dir)
+
+	clk := clock.NewMock(t)
+	ttl := 100
+
+	getAuthorizedEntriesAttempts := 0
+	entriesHaveBeenFetched := false
+
+	api := newMockAPI(t, &mockAPIConfig{
+		km: km,
+		getAuthorizedEntries: func(*mockAPI, int32, *entryv1.GetAuthorizedEntriesRequest) (*entryv1.GetAuthorizedEntriesResponse, error) {
+			// simulate 3 consecutive cache misses in server
+			if getAuthorizedEntriesAttempts < 3 {
+				getAuthorizedEntriesAttempts++
+				return &entryv1.GetAuthorizedEntriesResponse{
+					Entries: []*types.Entry{},
+				}, nil
+			}
+			entriesHaveBeenFetched = true
+			return makeGetAuthorizedEntriesResponse(t, "resp1", "resp2"), nil
+		},
+		batchNewX509SVIDEntries: func(*mockAPI, int32) []*common.RegistrationEntry {
+			return makeBatchNewX509SVIDEntries("resp1", "resp2")
+		},
+		svidTTL: ttl,
+		clk:     clk,
+	})
+
+	baseSVID, baseSVIDKey := api.newSVID(joinTokenID, 1*time.Hour)
+	cat := fakeagentcatalog.New()
+	cat.SetKeyManager(km)
+
+	c := &Config{
+		ServerAddr:       api.addr,
+		SVID:             baseSVID,
+		SVIDKey:          baseSVIDKey,
+		Log:              testLogger,
+		TrustDomain:      trustDomain,
+		Storage:          openStorage(t, dir),
+		Bundle:           api.bundle,
+		Metrics:          &telemetry.Blackhole{},
+		RotationInterval: time.Hour,
+		SyncInterval:     time.Hour,
+		Clk:              clk,
+		Catalog:          cat,
+		WorkloadKeyType:  workloadkey.ECP256,
+		SVIDStoreCache:   storecache.New(&storecache.Config{TrustDomain: trustDomain, Log: testLogger}),
+	}
+
+	m := newManager(c)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// set sync interval to a high value to proof that synchronizer retries sync
+	// with the lower default interval in case 0 entries are returned
+	m.c.SyncInterval = 100 * defaultSyncInterval
+
+	// initialize will generate the first attempt at fetching entries
+	if err := m.Initialize(ctx); err != nil {
+		t.Fatal(err)
+	}
+	initTime := clk.Now()
+	require.Equal(t, initTime, m.GetLastSync())
+
+	// keep clk moving so that synchronizer retries sync
+	go func(ctx context.Context) {
```

Comment:
I agree with changing the test to not have races anymore and submitted a new variant in my most recent commit, however, I did not adopt the proposed solution with the sync Channel. The issue I saw with it is that it would block the production code without something to drain the channel, as the test would do. Additionally, it would not really help in getting the "runSynchronizer" function past the "select" statement.
Instead I decided to reduce complexity and just use a real time clock paired with very low defaultSyncInterval settings for the test.